packages/contracts/src.ts/index.ts

```diff
@@ -649,12 +650,12 @@ export class BaseContract {
 
     // Wrapped functions to call emit and allow deregistration from the provider
     _wrappedEmits: { [ eventTag: string ]: (...args: Array<any>) => void };
-
-    constructor(addressOrName: string, contractInterface: ContractInterface, signerOrProvider?: Signer | Provider) {
+    // TODO: permute address first; make default value for address for hedera context
+    constructor(address: AccountLike| null, contractInterface: ContractInterface, signerOrProvider?: Signer | Provider) {
         logger.checkNew(new.target, Contract);
 
-        // @TODO: Maybe still check the addressOrName looks like a valid address or name?
-        //address = getAddress(address);
+        // @TODO: Maybe still check the addressOrName looks like a valid _address or name?
+        //_address = getAddress(_address);
```

Comment:
nit: we should remove those 2 lines and maybe do:
```TypeScript
this.address = address || "";
```
and remove the set on `706` line